pkgs/os-specific/linux/plymouth/default.nix

```diff
@@ -1,64 +1,119 @@
-{ stdenv, fetchurl, autoreconfHook, pkg-config, libxslt, docbook_xsl
-, gtk3, udev, systemd, lib
+{ lib
+, stdenv
+, fetchpatch
+, fetchFromGitLab
+, pkg-config
+, autoreconfHook
+, libxslt
+, docbook_xsl
+, intltool
+, gtk3
+, systemd
+, freetype
+, pango
+, cairo
+, libdrm
 }:
 
 stdenv.mkDerivation rec {
   pname = "plymouth";
-  version = "0.9.4";
+  version = "2020-12-07";
 
-  src = fetchurl {
-    url = "https://www.freedesktop.org/software/plymouth/releases/${pname}-${version}.tar.xz";
-    sha256 = "0l8kg7b2vfxgz9gnrn0v2w4jvysj2cirp0nxads5sy05397pl6aa";
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "plymouth";
+    repo = "plymouth";
+    rev = "c4ced2a2d70edea7fbb95274aa1d01d95928df1b";
```

Comment:
Any reason why this commit got picked? It has not tag and seems pretty random to me.

Reply:
It’s the commit where [!125 Use fallback image if BGRT is not supported](https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/125) was merged.